dummy/modowners_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"io/fs"
+	"log"
+	"strings"
+	"testing"
+	"testing/fstest"
+)
+
+func TestCommonElement(t *testing.T) {
+	for _, test := range []struct {
+		A      []string
+		B      []string
+		Result bool
+	}{
+		{nil, nil, false},
+		{[]string{"a"}, []string{"a"}, true},
+		{[]string{"a", "b"}, []string{"a"}, true},
+		{[]string{"a"}, []string{"b"}, false},
+	} {
+		if hasCommonElement(test.A, test.B) != test.Result {
+			t.Error(test)
+		}
+	}
+}
+
+func TestCheckTable(t *testing.T) {
+	for _, test := range []struct {
+		fakeMod       fs.FS
```

Comment:
I would make this optional then save the main one as a const so you don't have to define it for every test case

Reply:
Or you can save simply save the default case as a const and use it in each test.

Reply:
should i do this even if so far, i only have two test cases? one with a valid fake modfile and one with a invalid fake modfile

Reply:
Can always refactor it if you find yourself reusing the same file.

Reply:
i paired with serge today and we refactored the test table to have the first block be the structure of the test, second block with actual data (and no side effects), and third block have the logic!